### PR TITLE
tests: drivers: i2s: i2s_api: move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move drivers i2s_api tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

File: tests/drivers/i2s/i2s_api/src/test_i2s_dir_both_loopback.c [hunk 1]
```diff
@@ -18,7 +18,11 @@
 static ZTEST_DMEM const struct device *dev_i2s;
 static ZTEST_DMEM bool dir_both_supported;
 
-void test_i2s_dir_both_transfer_configure_0(void)
+#ifndef TEST_USERMODE
+ZTEST(i2s_dir_both_loopback_test_suite, test_i2s_dir_both_transfer_configure_0)
+#else
+ZTEST_USER(i2s_dir_both_user_loopback_test_suite, test_i2s_dir_both_transfer_configure_0)
+#endif
```
Comment (enjiamai) [inline tests/drivers/i2s/i2s_api/src/test_i2s_dir_both_loopback.c:25]: I remember @yperess said we just need to write the ZTEST_USER, don't need a #if/#define here.

**Review by enjiamai — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
Change to another coding style of the test

File: tests/drivers/i2s/i2s_api/src/main.c [hunk 2]
```diff
@@ -28,80 +28,264 @@ void test_main(void)
...
-	ztest_run_test_suite(i2s_dir_both_loopback_test);
-
-	ztest_test_suite(i2s_dir_both_states_test,
-		ztest_unit_test(test_i2s_dir_both_transfer_configure_1),
-		ztest_unit_test(test_i2s_dir_both_state_running_neg),
-		ztest_unit_test(test_i2s_dir_both_state_stopping_neg),
-		ztest_unit_test(test_i2s_dir_both_state_error_neg));
-	ztest_run_test_suite(i2s_dir_both_states_test);
-
-	/* Now run all tests in user mode */
```
Comment (stephanosio) [inline tests/drivers/i2s/i2s_api/src/main.c:69]: This is a rather strange way of testing user mode.

IMO, this should be handled through a separate userspace-enabled test (`drivers.i2s.userspace`) added to `testcase.yaml` that builds with `CONFIG_TEST_USERSPACE=y` -- that way, all these wrapper functions will not be necessary.

**Comment by enjiamai** [comment]
Hi @hunterhu3000 , I think you should rebase it instead of pull.

**Comment by hunterhu3000** [comment]
@stephanosio sorry for late, the PR is modified according to your suggestions.
@enjiamai The PR is rebased to latest
Thanks to review

**Review by enjiamai — APPROVED** [review]

File: tests/drivers/i2s/i2s_api/src/test_i2s_dir_both_loopback.c (hunk 1 above)
Comment (yperess) [reply tests/drivers/i2s/i2s_api/src/test_i2s_dir_both_loopback.c:25]: Not sure why this was resolved, yes, you don't need to have the `if/else` here

**Review by yperess — CHANGES_REQUESTED** [review]

**Pull request closed without merging**